content/en/docs/instrumentation/js/propagation.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Propagation
+description: Context propagation for the JS SDK
+aliases: [/docs/instrumentation/js/api/propagation]
+weight: 7
+---
+Propagation is the mechanism that moves data between services and processes.
+Although not limited to tracing, it is what allows traces to build causal
+information about a system across services that are arbitrarily distributed
+across process and network boundaries.
+
+## Context propagation with libraries
+
+For the vast majority of use cases, context propagation is done with
+instrumentation libraries.
+
+For example, if you have several Node.js services that communicate over HTTP,
+you can use the
+[express](https://www.npmjs.com/package/@opentelemetry/instrumentation-express)
+and [http](https://www.npmjs.com/package/@opentelemetry/instrumentation-http)
+instrumentation libraries to automatically propagate trace context across
+services for you.
+
+**It is highly recommend that you use instrumentation libraries to progagate
+context.** Although it is possible to propagate context manually, if your system
+uses libraries to communicate between services, use a matching instrumentation
+library to propagate context.
+
+Refer to [Libraries](/docs/instrumentation/js/libraries) to learn more about
+instrumentation libraries and how to use them.
+
+## Manual context propagation
+
+In some cases, it is not possible to propagate context with an instrumentation
+library. There may not be an instrumentation library that matches a library
+you're using to have services communicate with one another. Or you many have
+requirements that instrumentation libraries cannot fulfill, even if they exist.
+
+When you must propagate context manually, you can use the [context
+api](/docs/instrumentation/js/context).
+
+The following generic example demonstrates how you can propagate trace context
+manually.
+
+First, on the sending service, you'll need to inject the current `context`:
+
+```js
+// Sending service
+import { context, propagation, trace } from '@opentelemetry/api'
+const output = {}
+
+// Serialize the traceparent and tracestate from context into
+// an output object.
+//
+// This example uses the active trace context, but you can
+// use whatever context is appropriate to your scenario.
+propagation.inject(context.active(), output)
+
+const { traceparent, tracestate } = output
```

Comment:
There are no guarantees that the globally registered propagation would inject these fields. We should treat the output as opaque data transportation and send it as is.

Reply:
This is a question about which propagator you are using. IIUC, these fields are defined in W3C trace context spec. That is to say, when you are using the Jeager propagator as the globally registered one, you won't get these fields.

Reply:
Aha, gotcha, makes sense. Will think about how to document this.